pymemcache/client/base.py

```diff
@@ -794,7 +796,8 @@ def _fetch_cmd(self, name, keys, expect_cas):
                 return {}
             raise
 
-    def _store_cmd(self, name, values, expire, noreply, cas=None):
+    def _store_cmd(self, name, values, expire, noreply, cas=None,
+                   encoding='ascii'):
```

Comment:
Because `_store_cmd()` is an internal method, I think we should move `encoding` just before `cas` and _not_ give it a default value (so no keyword argument) because we're always providing a value for it from the public methods.

Reply:
_store_cmd() is being called by add()/ append()/ cas()/ prepend()/ replace()/ set()/ set_many(), from what you mentioned above, in order to make it consistency, should we add encoding for all these functions?

Reply:
> _store_cmd() is being called by add()/ append()/ cas()/ prepend()/ replace()/ set()/ set_many(), from what you mentioned above, in order to make it consistency, should we add encoding for all these functions?

Eek, you're right. Should `encoding` be a Client-wide setting (set in `__init__()`)? Or might we vary it based on each `set` operation?

Reply:
I'm pretty sure across remcached usage, we only use text protocol which is 'utf-8'. if we add encoding to `__init__()`, then in set and _store_cmd function, we won't need to add parameter encoding. Since the default encoding will be 'ascii', I don't think adding encoding into `__init__()` could cause any trouble for the current usage. If you think it's a good solution, i'll give it a try.

Reply:
> I'm pretty sure across remcached usage, we only use text protocol which is 'utf-8'. if we add encoding to `__init__()`, then in set and _store_cmd function, we won't need to add parameter encoding. Since the default encoding will be 'ascii', I don't think adding encoding into `__init__()` could cause any trouble for the current usage. If you think it's a good solution, i'll give it a try.

Assuming we're only going to apply this to value (and not key) encoding, I think setting this in `__init__()` will work well.

Keys are expected to be ASCII-encoded so that shouldn't be configurable: https://github.com/memcached/memcached/blob/78eb7701e0823643d693c1a7a6fd8a0c75db74d8/doc/protocol.txt#L612-L615

Reply:
yes, it's only for values. I'll work on it now.